pandas/_libs/algos_common_helper.pxi.in

```diff
@@ -22,54 +22,55 @@ ctypedef fused out_t:
 def diff_2d(ndarray[diff_t, ndim=2] arr,
             ndarray[out_t, ndim=2] out,
             Py_ssize_t periods, int axis):
+    cdef:
```

Comment:
I believe all you did was move this and not change conditions below but lmk if I am misreading

Reply:
The cdef here got moved up a few lines and the meat of the function got indented with an `else:` to avoid unreachable-code warnings